R/duplicates.R

```diff
@@ -0,0 +1,175 @@
+#' check if data.table contains duplicated data
+#' 
+#' An auxiliary function that checks for duplicates in the data.table (or its subset)
+#' 
+#' @param dt data.table
+#' @param col_names charvec with columns to be used for subsetting
+#' @examples
+#' dt <- data.table::data.table(a = c(1, 2, 3), b = c(3, 2, 2))
+#' has_dt_duplicated_rows(dt, "b")
+#' @return logical flag
+#' @keywords duplicates
+#'
+#' @export
+#' 
+has_dt_duplicated_rows <- function(dt, col_names = NULL) {
+  checkmate::assert_data_table(dt)
+  checkmate::assert_character(col_names, null.ok = TRUE)
+  
+  if (is.null(col_names)) {
+    anyDuplicated(dt) != 0
+  } else {
+    checkmate::assert_subset(col_names, colnames(dt))
+    anyDuplicated(dt, by = col_names) != 0
+  }
+  
+}
+
+#' get columns in the assay data required to have unique data
+#' 
+#' get columns in the assay data required to have unique (non-duplicated) data
+#' 
+#' @param dt data.table with assay data
+#' @examples
+#' sdata <- get_synthetic_data("finalMAE_small")
+#' smetrics_data <- convert_se_assay_to_dt(sdata[[1]], "Metrics")
+#' get_assay_uniq_cols(smetrics_data)
+#' @return charvec with columns required to have unique data
+#' @keywords duplicates
+#' @export
+#' 
+get_assay_uniq_cols <- function(dt) {
+  
+  checkmate::assert_data_table(dt)
+  col_ids <- get_settings_from_json(
+    "assay_dt_req_uniq_col_ids",
+    system.file(package = "gDRutils", "settings.json")
+  )
+  
+  # check with both pretiffied and unprettified version of ids
+  col_names_p <- unlist(get_prettified_identifiers(col_ids, simplify = FALSE))
+  col_names_up <- as.character(get_env_identifiers(col_ids, simplify = FALSE))
+  col_names <- unique(c(col_names_p, col_names_up))
+  
+  intersect(col_names, names(dt))
+}
+
+#' check if assay data contains duplicated data
+#' 
+#' An auxiliary function that checks for duplicates in the assay data
+#' 
+#' @param dt data.table with assay data
+#' 
+#' @return logical flag
```

Comment:
Could you add more description?
This boolean flag tells whether the dt test has any duplicates in the subset of columns defined by gDR as requiring unique values, right?

Reply:
👀